common/models/user.js

```diff
@@ -301,6 +301,25 @@ module.exports = function(User) {
     return fn.promise;
   };
 
+  User.observe('before delete', function(ctx, next) {
+    var AccessToken = ctx.Model.app.models.AccessToken;
```

Comment:
BTW the best long-term solution for this would be to implement support for [cascading delete](http://www.techonthenet.com/sql_server/foreign_keys/foreign_delete.php) in LoopBack relations. I am afraid that will require more work that we can afford right now. However, could you please open a feature request in juggler linked to this PR to keep track of this idea?